salt/modules/boto3_asg.py

```diff
@@ -0,0 +1,946 @@
+                if attr == "tags":
+                    _tags = []
+                    for tag in asg[keyname]:
+                        _tag = odict.OrderedDict()
+                        _tag["key"] = tag["Key"]
+                        _tag["value"] = tag["Value"]
+                        _tag["propagate_at_launch"] = tag["PropagateAtLaunch"]
+                        _tags.append(_tag)
+                    ret["tags"] = _tags
+                # Boto accepts a string or list as input for vpc_zone_identifier,
+                # but always returns a comma separated list. We require lists in
+                # states.
+                elif attr == "vpc_zone_identifier":
+                    ret[attr] = asg[keyname].split(",")
+                # convert SuspendedProcess objects to names
+                elif attr == "suspended_processes":
+                    ret[attr] = sorted([p["ProcessName"] for p in asg[keyname]])
+                elif attr == "placement_group":
+                    # Note that boto3 doesn't return a placement group, so skip it.
+                    continue
+                else:
+                    ret[attr] = asg[keyname]
+
+            # scaling policies
+            policies = conn.get_all_policies(as_group=name)
+            ret["scaling_policies"] = []
+            for policy in policies:
+                ret["scaling_policies"].append(
+                    dict([
+                        ("name", policy.name),
+                        ("adjustment_type", policy.adjustment_type),
+                        ("scaling_adjustment", policy.scaling_adjustment),
+                        ("min_adjustment_step", policy.min_adjustment_step),
+                        ("cooldown", policy.cooldown)
+                    ])
+                )
+            # scheduled actions
+            actions = conn.get_all_scheduled_actions(as_group=name)
+            ret['scheduled_actions'] = {}
+            for action in actions:
+                end_time = None
+                if action.end_time:
+                    end_time = action.end_time.isoformat()
+                ret['scheduled_actions'][action.name] = dict([
+                  ("min_size", action.min_size),
+                  ("max_size", action.max_size),
+                  # AWS bug
+                  ("desired_capacity", int(action.desired_capacity)),
+                  ("start_time", action.start_time.isoformat()),
+                  ("end_time", end_time),
+                  ("recurrence", action.recurrence)
+                ])
+            return ret
+        except boto.exception.BotoServerError as e:
+            if retries and e.code == 'Throttling':
+                log.debug('Throttled by AWS API, retrying in 5 seconds...')
+                time.sleep(5)
+                retries -= 1
+                continue
+            log.error(e)
+            return {}
+
+
+def create(name, launch_config_name, availability_zones, min_size, max_size,
```

Comment:
same here.. if we aren't upgrading it to use boto3, not sure it should be copied across.

Reply:
See above.